react/index.d.ts

```diff
@@ -270,9 +273,10 @@ declare namespace React {
     //
     // Event System
     // ----------------------------------------------------------------------
-    interface SyntheticEventBase<CURRENT, TARGET> {
+
+    interface SyntheticEvent<T> {
         bubbles: boolean;
-        currentTarget: EventTarget & CURRENT;
+        currentTarget: EventTarget & T;
```

Comment:
@vsiao :
The diff above change just restores the original definition of `SyntheticEvent`.

I agree with you that `T extends EventTarget` makes much more sense than just `T` but the changes required to implement that are quite extensive (and not directly related to this PR).

In order to introduce that refactor we would need to change:
  - Every event (e.g. `interface FormEvent<T> extends SyntheticEvent<T>`) - about 12 interfaces.
  - Every event handler (e.g. `type FormEventHandler<T> = EventHandler<FormEvent<T>>`) - about 12 type aliases
  - `DOMAttributes`
  - `HTMLAttributes` - requires also changing the types for `react-addons-linked-state-mixin`
  - etc.

I'm not against this change, but since it's so big, I believe we should create a separate PR for that.